packages/ra-core/src/controller/create/useCreateController.ts

```diff
@@ -60,13 +60,18 @@ export const useCreateController = <
         mutationOptions = {},
     } = props;
 
-    useAuthenticated({ enabled: !disableAuthentication });
     const resource = useResourceContext(props);
     if (!resource) {
         throw new Error(
             'useCreateController requires a non-empty resource prop or context'
         );
     }
+    useAuthenticated({ enabled: !disableAuthentication });
```

Comment:
The authentication check should be done before the authorization check. It does not make sense to verify the user rights if they are not authenticated. Therefore, you must read the `isPending` from `useAuthenticated`, and use it in the `enabled` flag of the `useRequireAccess`.

Same for other controllers.